src/Simplify.elm

```diff
@@ -5887,7 +5905,10 @@ dictMapChecks =
 
 dictIntersectChecks : IntoFnCheck
 dictIntersectChecks =
-    collectionIntersectChecks dictCollection
+    intoFnChecksFirstThatConstructsError
+        [ collectionIntersectChecks dictCollection
+        , whenArgumentsAreEqualReturnLastCheck
```

Comment:
Why not include `whenArgumentsAreEqualReturnLastCheck` directly in `collectionIntersectChecks`?